alphadia/fdrx/models/two_step_classifier.py

```diff
@@ -83,52 +90,94 @@ def fit_predict(
             DataFrame containing predictions and q-values
 
         """
+        logger.info("=== Starting training of TwoStepClassifier ===")
+
         df = self._preprocess_data(df, x_cols)
         best_result = None
-        best_precursor_count = -1
+
+        previous_target_count_after_first_clf = -1
+        previous_target_count_after_second_clf = -1
 
         for i in range(self._max_iterations):
+            logger.info(f"Starting iteration {i + 1} / {self._max_iterations}.")
+
+            # extract preselction using first classifier if it is fitted
             if self.first_classifier.fitted and i > 0:
-                df_train, df_predict = self._apply_filtering_with_first_classifier(
+                df_train = self._apply_filtering_with_first_classifier(
                     df, x_cols, group_columns
                 )
+                df_predict = df_train  # using the same df for training and predicting, unlike in the following else block.
+
+                previous_target_count_after_first_clf = get_target_count(df_train)
```

Comment:
Should we add a log statement like "applying linear classifier at 60% FDR - xxx targets and xxx decoys"?